sdk/program/src/sysvar/instructions.rs

```diff
@@ -2,30 +2,40 @@
 //! This account contains the serialized transaction instructions
 use crate::{
     account_info::AccountInfo,
+    impl_sysvar_get,
     instruction::{AccountMeta, Instruction},
     program_error::ProgramError,
     pubkey::Pubkey,
     sanitize::SanitizeError,
     serialize_utils::{read_pubkey, read_slice, read_u16, read_u8},
+    sysvar::Sysvar,
 };
+
 #[cfg(not(target_arch = "bpf"))]
 use {
     crate::serialize_utils::{append_slice, append_u16, append_u8},
     bitflags::bitflags,
 };
 
 // Instructions Sysvar, dummy type, use the associated helpers instead of the Sysvar trait
+#[derive(Serialize, Clone, Deserialize, Debug, Default, PartialEq)]
 pub struct Instructions();
```

Comment:
I can't wrap my head around how instruction introspection is working (apparently it does something extra at runtime?). It seems to be more complicated than implementing a trait to the dummy struct.